DATA_LAYER_INTEGRATION.md

```diff
@@ -269,13 +269,142 @@ The JSON rendering of a Core Component exposes a `dataLayer` property that is po
 
 To automatically add a custom component to the data layer:
 1. Define the properties of the custom component model that needs to be tracked.
-1. Add a component ID to the the custom component HTL.
 1. Add the `data-cmp-data-layer` attribute to the custom component HTL. E.g. `data-cmp-data-layer="${mycomponent.data.json}"`.
 
 To automatically make the data layer trigger a `cmp:click` event each time a specific element of the custom component is clicked:
-1. In the custom component HTL add the `data-cmp-clickable` attribute to the element to be tracked.
+in the custom component HTL add the `data-cmp-clickable` attribute to the element to be tracked.
 
 The `data-cmp-data-layer-enabled` attribute can be queried client side to check if the data layer is enabled.
 
+### Examples
+
+This section provides a few examples how to make a custom component write data to the data layer.
+
+#### Pre-requisite: create a HelloWorld component
+
+Create a `HelloWorld` model and HTL script that prints "Hello World!" to the page:
+
+`HelloWorld` model:
+
+```
+...
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.models.annotations.Model;
+import com.adobe.cq.wcm.core.components.internal.models.v1.AbstractComponentImpl;
+...
+
+@Model(adaptables = SlingHttpServletRequest.class)
+public class HelloWorld extends AbstractComponentImpl {
```

Comment:
`AbstractComponentImpl` is an internal class, it would probably make more sense to to extend from 
`com.adobe.cq.wcm.core.components.models.Component`.

Reply:
addressed in commit https://github.com/adobe/aem-core-wcm-components/pull/1215/commits/2c7c6e594f6a35e14d19a0215406b4a871a49161